platform/measurement_utils.cpp

```diff
@@ -49,7 +53,37 @@ string FormatDistanceImpl(Units units, double m, string const & low, string cons
   {
     double const highV = m / highF;
     // For distances of 10.0 high units and over round to a whole number, e.g. 9.98 -> 10, 10.9 -> 11
-    return ToStringPrecision(highV, round(highV * 10) / 10 >= 10.0 ? 0 : 1) + " " + high;
+
+    int precision = round(highV * 10) / 10 >= 10.0 ? 0 : 1;
+
+    string out = ToStringPrecision(highV, precision);
+
+    if (precision)
+    {
+      // Value with decimals. Set locale decimal separator
+      if (loc.m_decimal_separator != '.')
+      {
+          std::replace(out.begin(), out.end(), '.', loc.m_decimal_separator);
```

Comment:
Yes, instead of `std::replace()` we could assume that the decimal separator is at `out.size() - 2`. I will change the code accordingly...
And yes, 3.0 is printed as 3.0

Reply:
And regarding the unit tests, please confirm that the idea is to add to `measurement_tests.cpp` tests of the `FormatDistanceImpl()` function to check the functionality of the decimal and grouping separators replacement based on different locales.